compiler/src/dotty/tools/dotc/inlines/Inlines.scala

```diff
@@ -199,14 +199,7 @@ object Inlines:
     var unapplySym1: Symbol = NoSymbol // created from within AnonClass() and used afterwards
 
     val newUnapply = AnonClass(ctx.owner, List(defn.ObjectType), sym.coord) { cls =>
-      val targs = fun match
-        case TypeApply(_, targs) => targs
-        case _ => Nil
-      val unapplyInfo = sym.info match
-        case info: PolyType => info.instantiate(targs.map(_.tpe))
```

Comment:
Why isn't this needed anymore? The comment above could maybe explain how type parameters are handled.

Reply:
That's handled by `fun.tpe.widen` (line 202)

Reply:
I added a comment below and managed to simplify the leading implicits logic thanks to the use of `fun.tpe.widen`.